test/kitchen-sink.json

```diff
@@ -1 +1 @@
":14},"value":"count"},"arguments":null,"directives":null,"selectionSet":null}]}},{"kind":"Field","loc":{"start":7,"end":8},"alias":null,"name":{"kind":"Name","loc":{"start":7,"end":19},"value":"likeSentence"},"arguments":null,"directives":null,"selectionSet":{"kind":"SelectionSet","loc":{"start":20,"end":8},"selections":[{"kind":"Field","loc":{"start":9,"end":13},"alias":null,"name":{"kind":"Name","loc":{"start":9,"end":13},"value":"text"},"arguments":null,"directives":null,"selectionSet":null}]}}]}}]}}]}},{"kind":"FragmentDefinition","loc":{"start":1,"end":2},"name":{"kind":"Name","loc":{"start":10,"end":14},"value":"frag"},"typeCondition":{"kind":"NamedType","loc":{"start":18,"end":24},"name":{"kind":"Name","loc":{"start":18,"end":24},"value":"Friend"}},"directives":null,"selectionSet":{"kind":"SelectionSet","loc":{"start":25,"end":2},"selections":[{"kind":"Field","loc":{"start":3,"end":54},"alias":null,"name":{"kind":"Name","loc":{"start":3,"end":6},"value":"foo"},"arguments":[{"kind":"Argument","loc":{"start":7,"end":18},"name":{"kind":"Name","loc":{"start":7,"end":11},"value":"size"},"value":{"kind":"Variable","loc":{"start":13,"end":18},"name":{"kind":"Name","loc":{"start":13,"end":18},"value":"size"}}},{"kind":"Argument","loc":{"start":20,"end":27},"name":{"kind":"Name","loc":{"start":20,"end":23},"value":"bar"},"value":{"kind":"Variable","loc":{"start":25,"end":27},"name":{"kind":"Name","loc":{"start":25,"end":27},"value":"b"}}},{"kind":"Argument","loc":{"start":29,"end":53},"name":{"kind":"Name","loc":{"start":29,"end":32},"value":"obj"},"value":{"kind":"ObjectValue","loc":{"start":34,"end":53},"fields":[{"kind":"ObjectField","loc":{"start":35,"end":39},"name":{"kind":"Name","loc":{"start":35,"end":38},"value":"key"},"value":{"kind":"StringValue","loc":{"start":38,"end":39},"value":"value"}}]}}],"directives":null,"selectionSet":null}]}},{"kind":"OperationDefinition","loc":{"start":1,"end":2},"operation":"query","name":null,"variableDefinitions":null,"directives":null,"selectionSet":{"kind":"SelectionSet","loc":{"start":1,"end":2},"selections":[{"kind":"Field","loc":{"start":3,"end":54},"alias":null,"name":{"kind":"Name","loc":{"start":3,"end":10},"value":"unnamed"},"arguments":[{"kind":"Argument","loc":{"start":11,"end":23},"name":{"kind":"Name","loc":{"start":11,"end":17},"value":"truthy"},"value":{"kind":"BooleanValue","loc":{"start":19,"end":23},"value":true}},{"kind":"Argument","loc":{"start":25,"end":38},"name":{"kind":"Name","loc":{"start":25,"end":31},"value":"falsey"},"value":{"kind":"BooleanValue","loc":{"start":33,"end":38},"value":false}},{"kind":"Argument","loc":{"start":40,"end":53},"name":{"kind":"Name","loc":{"start":40,"end":47},"value":"nullish"},"value":{"kind":"NullValue","loc":{"start":49,"end":53}}}],"directives":null,"selectionSet":null},{"kind":"Field","loc":{"start":3,"end":8},"alias":null,"name":{"kind":"Name","loc":{"start":3,"end":8},"value":"query"},"arguments":null,"directives":null,"selectionSet":null}]}}]}
```

Comment:
hand-vetted this. we add a pair of `"typeCondition": null` lines that weren't there before, bringing us more into line with graphql-js output. (remaining differences: graphql-js prints empty containers instead of nulls for at least some nullable plural fields, and graphql-js locations are offset-based instead of line-based)